client/backwards_compatibility_test.go

```diff
@@ -131,6 +131,70 @@ func Test0Dot1RepoFormat(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// We can read and publish from notary0.3 repos
+func Test0Dot3RepoFormat(t *testing.T) {
+	// make a temporary directory and copy the fixture into it, since updating
+	// and publishing will modify the files
+	tmpDir, err := ioutil.TempDir("", "notary-backwards-compat-test")
+	defer os.RemoveAll(tmpDir)
+	require.NoError(t, err)
+	require.NoError(t, recursiveCopy("../fixtures/compatibility/notary0.3", tmpDir))
+
+	gun := "docker.com/notary0.3/tst"
+	passwd := "password"
+
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	repo, err := NewNotaryRepository(tmpDir, gun, ts.URL, http.DefaultTransport,
+		passphrase.ConstantRetriever(passwd), trustpinning.TrustPinConfig{})
+	require.NoError(t, err, "error creating repo: %s", err)
+
+	// targets should have 1 target, and it should be readable offline
+	targets, err := repo.ListTargets()
+	require.NoError(t, err)
+	require.Len(t, targets, 3)
+
+	// ok, now that everything has been loaded, verify that the fixture is valid
+	requireValidFixture(t, repo)
+
+	// delete the timestamp metadata, since the server will ignore the uploaded
+	// one and try to create a new one from scratch, which will be the wrong version
+	require.NoError(t, repo.fileStore.Remove(data.CanonicalTimestampRole))
+
+	// rotate the timestamp key, since the server doesn't have that one
+	err = repo.RotateKey(data.CanonicalTimestampRole, true)
+	require.NoError(t, err)
+
+	require.NoError(t, repo.Publish())
+
+	targets, err = repo.ListTargets()
+	require.NoError(t, err)
+	require.Len(t, targets, 5)
+	// the changelist target/releases delegation will get published with the above publish
+	delegations, err := repo.GetDelegationRoles()
+	require.NoError(t, err)
+	require.Len(t, delegations, 1)
+	require.Equal(t, "targets/releases", delegations[0].Name)
+
+	// Also check that we can add/remove keys by rotating keys
+	oldTargetsKeys := repo.CryptoService.ListKeys(data.CanonicalTargetsRole)
+	require.NoError(t, repo.RotateKey(data.CanonicalTargetsRole, false))
+	require.NoError(t, repo.Publish())
+	newTargetsKeys := repo.CryptoService.ListKeys(data.CanonicalTargetsRole)
+
+	require.Len(t, oldTargetsKeys, 1)
+	require.Len(t, newTargetsKeys, 1)
+	require.NotEqual(t, oldTargetsKeys[0], newTargetsKeys[0])
+
+	// rotate the snapshot key to the server and ensure that the server can re-generate the snapshot
+	// and we can download the snapshot
+	require.NoError(t, repo.RotateKey(data.CanonicalSnapshotRole, true))
+	require.NoError(t, repo.Publish())
+	err = repo.Update(false)
+	require.NoError(t, err)
+}
```

Comment:
Since we're migrating these, could we add a test for the 0.3 and the 0.1 repos to ensure that once a notary repo is created, all the keys are migrated to the new format?